api/lnd/index.js

```diff
@@ -10,6 +11,11 @@ const lnd = global.lnd || authenticatedLndGrpc({
   macaroon: process.env.LND_MACAROON,
   socket: process.env.LND_SOCKET
 }).lnd
+installLNDK(lnd, {
```

Comment:
You can then also remove all these checks:

```js
const lndk = lnd?.lndk
if (!lndk) throw new Error('lndk not installed, please use installLNDK')
```

Reply:
Since the Tor patch, we’ve been creating this in /lib/lnd.js, but I understand your concerns.

I am not a fan of the proposed alternative of passing it through context, for the following reasons:
- it adds yet another field that we need to remember to pass through all the functions
- it depends on lnd being also passed for fee estimation
- it is error-prone and defeats the purpose of having a unified Bolt interface
- lndk doesn't implement the full bolt12 standard and there are plans to implement bolt12 natively in lnd (see [1]). So, if we make all the functions depends already only on lnd, we won't have to refactor anything outside of the bolt12 interface to switch to the native lnd implementation.

<details>
  <summary>[1]</summary>
https://github.com/lightningnetwork/lnd/issues/5594
https://github.com/lightningnetwork/lnd/pull/9369

![image](https://github.com/user-attachments/assets/9efb7682-3ea2-4de8-98b0-fad3d7108a49)
</details>
 
So, to address your concerns, I’ve changed the code to use a WeakMap to maintain lndk instances until their corresponding lnd is GCed. This should address your main concern that, as far as i understand, is that in the remote eventuality that the lnd client decides to export a service called lndk, the previous code could end up shadowing it.